src/controllers/highlights/HighlightBadge.cpp

```diff
@@ -0,0 +1,99 @@
+#include "HighlightBadge.hpp"
+#include "singletons/Resources.hpp"
+
+namespace chatterino {
+
+QColor HighlightBadge::FALLBACK_HIGHLIGHT_COLOR = QColor(127, 63, 73, 127);
+
+bool HighlightBadge::operator==(const HighlightBadge &other) const
+{
+    return std::tie(this->badgeName_, this->badgeVersion_, this->displayName_,
+                    this->hasSound_, this->hasAlert_, this->soundUrl_,
+                    this->color_) ==
+           std::tie(other.badgeName_, other.badgeVersion_, other.displayName_,
+                    other.hasSound_, other.hasAlert_, other.soundUrl_,
+                    other.color_);
+}
+
+HighlightBadge::HighlightBadge(const QString &badgeName,
+                               const QString &badgeVersion,
+                               const QString &displayName, bool hasAlert,
+                               bool hasSound, const QString &soundUrl,
+                               QColor color)
+    : badgeName_(badgeName)
+    , badgeVersion_(badgeVersion)
```

Comment:
What's the purpose of `badgeVersion`?

Reply:
`badgeVersion` is the version component of the badge sent in IRC tags: https://dev.twitch.tv/docs/irc/tags#globaluserstate-twitch-tags (under 'badges')

Both the name and version of a badge must match for it to be highlighted

Reply:
> Currently this is used only for subscriber, to indicate the exact number of months the user has been a subscriber. This number is finer grained than the version number in badges. For example, a user who has been a subscriber for 45 months would have a badge-info value of 45 but might have a badges version number for only 3 years.

We don't have the ability to highlight on subscribers, and we don't have a way in the UI to highlight based on the version, so this feels like a future failure case.

Example: If someone has `vip` highlighted for version `1` (which the user has no control over) then Twitch updates the version in the IRC tags to update the image, then the highlight doesn't go through anymore.